### PR TITLE
Route: Added domain second level name (%sld%)

### PR DESCRIPTION
**Pull request by JanGalek**
**REBASE**
better use for location by tld.

**example:**
`$router[] = new Route('//[!www.]%sld%.<locale cz|sk>/radce-podekovani','Radce:podekovani');`

### CODE REVIEW

**Comment by dg** [comment]
Domain name is `example.com`, TLD is `com`, I am not sure which name to use, but %domainName% is misleading.  https://en.wikipedia.org/wiki/Domain_name

**Comment by GcoreCZ** [comment]
sorry, its Second-level of domain.
Should I change it ?

File: src/Application/Routers/Route.php [hunk 1]
```diff
@@ -155,6 +155,7 @@ public function match(Nette\Http\IRequest $httpRequest)
 				'/%basePath%/' => preg_quote($url->getBasePath(), '#'),
 				'%tld%' => preg_quote($host[0], '#'),
 				'%domain%' => preg_quote(isset($host[1]) ? "$host[1].$host[0]" : $host[0], '#'),
+				'%sld%'=>preg_quote(isset($host[1]) ? "$host[1]" : $host[0], '#'),
```
Comment (dg) [inline src/Application/Routers/Route.php]: IMHO should be `'%sld%' => preg_quote($host[1], '#'),`

File: src/Application/Routers/Route.php [hunk 2]
```diff
@@ -396,6 +397,7 @@ public function constructUrl(Application\Request $appRequest, Nette\Http\Url $re
 				'/%basePath%/' => $refUrl->getBasePath(),
 				'%tld%' => $host[0],
 				'%domain%' => isset($host[1]) ? "$host[1].$host[0]" : $host[0],
+				'%sld%' => isset($host[1]) ? "$host[1]" : $host[0],
```
Comment (dg) [inline src/Application/Routers/Route.php]: `'%sld%' => isset($host[1]) ? $host[1] : '',

File: src/Application/Routers/Route.php (hunk 1 above)
Comment (GcoreCZ) [inline src/Application/Routers/Route.php]: you have right, Ill repair it ;)

**Comment by Majkl578** [comment]
How about e.g. co.uk?

**Comment by dg** [comment]
@Majkl578  It is out of scope of this PR

**Comment by dg** [comment]
@GcoreCZ please add a test

**Comment by JanTvrdik** [comment]
@Majkl578 Impossible to solve, feel free to correct me with PR =)

**Comment by GcoreCZ** [comment]
@dg Ok

File: src/Application/Routers/Route.php [hunk 3]
```diff
@@ -155,6 +155,7 @@ public function match(Nette\Http\IRequest $httpRequest)
 				'/%basePath%/' => preg_quote($url->getBasePath(), '#'),
 				'%tld%' => preg_quote($host[0], '#'),
 				'%domain%' => preg_quote(isset($host[1]) ? "$host[1].$host[0]" : $host[0], '#'),
+				'%sld%'=>preg_quote(isset($host[1]) ? "$host[1]" : '', '#'),
```
Comment (hrach) [inline src/Application/Routers/Route.php]: spaces round =>

**Comment by GcoreCZ** [comment]
Test added, My first again :), hopefully will be ok, I took it from basic.

File: src/Application/Routers/Route.php (hunk 3 above)
Comment (GcoreCZ) [inline src/Application/Routers/Route.php]: ok, sorry

Comment (dg) [inline src/Application/Routers/Route.php]: And no qoutes around $host[1]

**Comment by GcoreCZ** [comment]
Ok, so can I merge commits ? :)

**Comment by GcoreCZ** [comment]
@Majkl578 co.uk is tld ;),but I should test it too.

**Comment by GcoreCZ** [comment]
@dg sorry for this long time and too commits,etc.. Next PR will be better. I promise.

**Comment by Majkl578** [comment]
> co.uk is tld ;),but I should test it too.

It's not. `.uk` is TLD, `.co.uk` is not, it's regular second level domain.
https://en.wikipedia.org/wiki/.co_(second-level_domain)

File: src/Application/Routers/Route.php (hunk 3 above)
Comment (dg) [inline src/Application/Routers/Route.php]: @GcoreCZ  This is still not fixed

Comment (GcoreCZ) [inline src/Application/Routers/Route.php]: @dg sorry, its ok now.

**Comment by GcoreCZ** [comment]
Is everything ok ? :)

**Comment by dg** [comment]
Do you know how to squash it to single commit?

**Comment by GcoreCZ** [comment]
sorry, but no, I'm learning with git still, so and I use netbeans and Iam not sure if it have option for it.

**Comment by GcoreCZ** [comment]
OK squashed :)

**Comment by dg** [comment]
Thanks, merged 209e227968a7970a96e5583c7e4ad05d05fc3979 :-) (I modified a little test)

**Comment by JanGalek** [comment]
@dg np, thanks for modifield, Can I add this to documentation or you want wait to next release version ? :)

**Comment by dg** [comment]
Wow, you can add it, that would be cool.

**Comment by JanGalek** [comment]
@dg Ok I sent PR https://github.com/nette/web-content/pull/425 ;)

**Pull request closed without merging**